internal/observability/call.go

```diff
@@ -434,7 +469,18 @@ func (c call) logStreamEvent(err error, success bool, succMsg, errMsg string, ex
 	if success {
 		ce = c.edge.logger.Check(c.levels.success, succMsg)
 	} else {
-		ce = c.edge.logger.Check(c.levels.failure, errMsg)
+		var lvl zapcore.Level
+		if c.levels.useApplicationErrorFailureLevels {
+			lvl = c.levels.failure
+		} else {
+			lvl = c.levels.serverError
+			code := yarpcerrors.FromError(err).Code()
+			if fault := faultFromCode(code); fault == clientFault {
+				lvl = c.levels.clientError
```

Comment:
Minor comment: This code looks pretty much similar to Line206-213, is it possible to move the common code in to a method and use that method to avoid code duplication?

Reply:
Do you have any recommendation for this? Except creating a wrapper around (which seems overkill)

```
if fault := faultFromCode(code); fault == clientFault {
 				lvl = c.levels.clientError
}
```
I do not see something else. It should be noted that line 206:213, a pointer of code is being used and not a ref